### PR TITLE
Update Page.php

### PR DESCRIPTION
**Pull request by ob1201**
I am targeting this branch, because fails on site load

Addresses #965

## Changelog
```markdown
### Fixed
missing string wrapper around trim
```

### CODE REVIEW

**Comment by greg0ire** [comment]
Please format your commit message according to our rules : 

- The commit message subject must be less than 50 characters and tell us *what* you did.
- The commit message body should tell us *why* you did it. It is optional but highly recommended.

Bad example :

```
Fixed bug #989 by removing call to defraculate()
```

Good example:

```
Remove call to defraculate()

Calling this function caused a bug because it interferes 
with calls to getSchmeckles().
Fixes #989
```

More details in CONTRIBUTING.md

File: src/Model/Page.php [hunk 1]
```diff
@@ -306,7 +306,7 @@ public function getName()
      */
     public function setSlug($slug)
     {
-        $this->slug = self::slugify(trim($slug));
+        $this->slug = self::slugify(trim((string)$slug));
```
Comment (greg0ire) [inline src/Model/Page.php]: missing space between the cast and the variable, this is why the build fails. Please amend your commit.

Comment (ob1201) [reply src/Model/Page.php]: this is what you suggested
```
so the fix would be https://github.com/sonata-project/SonataPageBundle/blob/3.x/src/Model/Page.php#L309, adding (string) before $slug
```
that's simply what was done above, no?

Comment (kunicmarko20) [reply src/Model/Page.php]: @ob1201 you just need to add ` `  between `(string)` and `$slug` so you get `$this->slug = self::slugify(trim((string) $slug));`

File: src/Model/Page.php [hunk 2]
```diff
@@ -306,7 +306,7 @@ public function getName()
      */
     public function setSlug($slug)
     {
-        $this->slug = self::slugify(trim($slug));
+        $this->slug = self::slugify(trim((string) $slug));
```
Comment (greg0ire) [inline src/Model/Page.php]: I hope nothing breaks if we change a slug from `null` to `''`. BTW, when does this occur? Can't see why the slug of a page would be `null` or `''`

Comment (jordisala1991) [reply src/Model/Page.php]: trim(null) = '' AFAIK

Comment (kunicmarko20) [reply src/Model/Page.php]: https://3v4l.org/71CT5 vs https://3v4l.org/ufpTl, not sure this is our problem?

Comment (jordisala1991) [reply src/Model/Page.php]: https://3v4l.org/3egP6

**Comment by greg0ire** [comment]
Thanks @ob1201 !

**Comment by greg0ire** [comment]
Wait what version of php are you using to get this warning? hhvm?

**Comment by ob1201** [comment]
@greg0ire 
```
php -v
PHP 7.2.1-dev (cli) (built: Jan  8 2018 15:40:39) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
   with Zend OPcache v7.2.1-dev, Copyright (c) 1999-2018, by Zend Technologies
```

updating to this^^
```
composer require sonata-project/page-bundle:3.x-dev
```

_does_ fix the trim error, both at console, and on site frontpage -- where I'd been seeing it

**Comment by greg0ire** [comment]
`7.2.1-dev`

You're using a dev branch of php?!

**Comment by ob1201** [comment]
This is fully reproducible with 7.2.1 *release* as well

**Comment by greg0ire** [comment]
Must have been fix in between then: 

```PHP 7.2.2 (cli) (built: Feb  6 2018 22:51:22) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.2, Copyright (c) 1999-2018, by Zend Technologies
bash-4.4$ php -r "var_dump(trim(null));"
string(0) ""

```

**Comment by greg0ire** [comment]
Can't reproduce here either: https://3v4l.org/ufoMl am I doing the wrong test?

**Pull request merged**